### ISSUE TITLE
Poco::Application constructor changes root Logger

### ISSUE THREAD

**Issue by jleben** [pocoproject/poco#1674]
The channel and level set on the root logger (`Logger::root()`) before creating an instance of Poco::Application are overridden when the application is constructed.

This means that setting up logging in the "main" function as suggested in the introductory slides (https://pocoproject.org/slides/110-Logging.pdf), before Poco::Application is created, has no effect.

Moreover, the documentation of Poco::Application (https://pocoproject.org/docs/Poco.Util.Application.html) says nothing about modifications to the root logger. Instead, it only mentions loggers "ApplicationStartup", "Application", and loggers defined in configuration. If there is no configuration, any modification to the root logger is unexpected.

This also means that instantiating a Poco::Application affects all loggers created after, even if they are not descendants of the application logger, which is undesirable.

I propose to change the behavior so that the root logger remains unmodified after a Poco::Application is instantiated as well as after it is initialized, unless the configuration explicitly requests modifications.

**Comment by aellerton** [comment]
I recognize this is an old issue but want to post here in case it helps anyone else.

I just spent some hours trying to figure out why logging from configuration files wasn't being picked up. Very frustrating!

If my service was called `foo.exe` and my config was `foo.ini`, it just wouldn't pick up my changes. I tried `foo.properties` in case it was that; I tried removing the root level `[logging]` from the INI file, all to no avail.

I wrote a utility function to dump out the configuration, and it confirmed it had loaded ok. I read and re-read the logging doc (both PDF and in the source). Still nothing.

The key was two things.

First, my service did this:

```
    void initialize(Application& self)
    {
        // WRONG
        ServerApplication::initialize(self);
        loadConfiguration(); // load default configuration files, if present
    }
```

The call to `LoadConfiguration()` _must_ come first:

```
    void initialize(Application& self)
    {
        // CORRECT
        loadConfiguration(); // load default configuration files, if present
        ServerApplication::initialize(self);
    }
```

Secondly, it seems like the application logger must be named, and that name mentioned in the config. Here's my working INI log config:

```ini
[logging]
formatters.f2.class = PatternFormatter
formatters.f2.pattern = %L%Y-%m-%d-T%H-%M-%S.%i %s [%q] %t
formatters.f2.times = UTC
channels.console.class = ConsoleChannel
channels.console.formatter = f2
loggers.main.channel = console
loggers.main.level = debug
loggers.main.name = daemon-logger

[application]
logger = daemon-logger
```

Note how I explicitly name the logger in two places!

Circling back to the origin comment by @jleben , I wondered if the `Poco::Application::setup()` call, which appears to blank out the root logger config, was the culprit, but `setup()` is called in the constructor, and really just puts a simple "just write to console" configuration into the logger so something appears when a log function is called.

In other words, there is no need to change the current behavior.
